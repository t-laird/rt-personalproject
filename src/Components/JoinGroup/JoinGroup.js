import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../Actions';
import { NavLink } from 'react-router-dom';
import validateGroup from '../../helpers/validateGroup';



class JoinGroup extends Component {
	constructor() {
		super();

		this.state = {
			passphrase: '',
			message: `You haven't joined a group yet.  And you should!  Enter your group passphrase below:`
		}
	}

	componentDidMount = () => {
		if (this.props.group.group_name) {
			this.setState({message: `You are a member of: ${this.props.group.group_name}`})
		} else if (this.props.user.group_id) {
			this.setState({message: 'You are a member of a group!'})
		}
	}

	handleChange = (event) => {
		const { target } = event;
		const { name, value } = target;

		this.setState({[name]: value})
	}

	joinGroup = async (event) => {
		event.preventDefault();
		const response = await validateGroup(this.state.passphrase, this.props.user.user_id)
		this.props.updateUser(response[0])
	}

	render() {
		return (
			<div>
				<h4>{this.state.message}</h4>
				<form>
					<input 
						value={this.state.passphrase}
						onChange={this.handleChange}
						placeholder='passphrase' 
						name='passphrase' />
					<button
						onClick={this.joinGroup}
					>submit</button>
				</form>

				<NavLink to='/creategroup'>Create Group</NavLink>

			</div>
		) 
	}
}

const mapStateToProps = ( store ) => ({
	user: store.User,
	group: store.Group
})

const mapDispatchToProps = dispatch => ({
	updateUser: user => {
		dispatch(actions.updateUser(user))
	}
});

export default connect(mapStateToProps, mapDispatchToProps)(JoinGroup);