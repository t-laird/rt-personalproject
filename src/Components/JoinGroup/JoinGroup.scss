@import '../../Variables.scss';

.join-group-component {

  @include center-box;
  background-color: $c6;

  .join-group-header {
    @include box-header;
    background-color: $c1;

    h4 {
      color: $c7;
      font-weight: 900;
      text-align: center;
    }
  }

  .join-container {
    padding: 24px 40px;

    a {
      font-size: 24px;
      color: $c3;
      font-weight: 900;
      position: relative;
      top: 9px;
    }

    h5 {
      color: $c3;
      font-weight: 900;
    }
  }

  .current-group, .passphrase {
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;
  }

  .current-group {
    h4 {
      color: $c1;
      text-align: center;
      width: 220px;
      height: 60px;
      border-bottom: 8px solid $c4;
      background-color: $c5;
    }
  }

  .join-group-last {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    margin-bottom: 30px;
  }

  .join-group-last {
    .arrow-graphic {
      width: 85px;
      margin-left: 20px;
    }
  }

  .join-group-last h5 {
    width: 405px;
  }
/*
	.link-holder {
		width: 770px;
		margin: 0 auto 30px;
		padding: 20px 25px;
	}*/

	/*h3 {
		color: $blue-text;
    line-height: 1.3;
    text-align: center;
    width: 100%;
    margin-top: 10px;
    a {
      color: inherit;
      font-size: inherit;
      font-weight: inherit;
      text-decoration: underline;
    }
	}*/

/*	form {
		width: 384px;
		display: flex;
		justify-content: space-between;
		margin: auto;
	}*/

	input, button {
		overflow: visible;
    border: none;
    height: 60px;
    font-size: 24px;
    margin-top: 10px;
    text-align: center;
	}

	input {
		width: 220px;
		background-color: $c7;
    border-bottom: 8px solid $c5;
		color: $c3;

		&::placeholder {
			color: $c1;
      font-weight: 900;
		};
	}

  button {
    height: 52px;
    width: 149px;
    color: #fff;
    background-color: $c1;
    cursor: pointer;
    transition: background-color .15s ease-in-out;
    padding-top: 2px;
    margin: auto;
    border: none;
    display: block;
    font-size: 20px;

    &:hover {
      background-color: darken($c1, 12%);
    }
  }

}

.link-holder {
  @include center-box;

	.create-group-link {
		color: $c1;
		font-weight: 900;
		text-align: right;
		display: block;
		transition: color .15s ease-in-out;
    font-size: 20px;
	}

	.create-group-link:hover {
		color: darken($c1, 12%);
	}
}
