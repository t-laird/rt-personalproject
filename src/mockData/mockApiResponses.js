export const getGroupSettingsApiResponse = {
  status: 'success',
};

export const getGroupTransactionDataApiResponse = {
	status: 'success'
};