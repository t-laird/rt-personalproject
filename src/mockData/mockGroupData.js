export const mockGroupData = {
	administrator_id: 95,
	created_date: "2017-10-18T18:34:30.017Z",
	group_id: 1,
	group_name: 'Turing',
	group_passphrase: "9139z",
	weekly_points: 100
}