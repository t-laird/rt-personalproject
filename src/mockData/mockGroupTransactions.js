const mockGroupTransactions = [
  {transactions: [
    {
      created_time: 1514939019586,
      event_id: 9,
      event_time: "2018-01-03T00:23:39.627Z",
      group_id: 1,
      note: "You totally killed it on your lightning talk this morning!  I loved it.",
      point_value: 15,
      receive_id: 22,
      received_name: "Scott Adams",
      send_id: 24,
      send_name: "Jim Davis"
    },
    {
      created_time: 1514939019586,
      event_id: 2,
      event_time: "2018-01-03T00:23:39.612Z",
      group_id: 1,
      note: "Thanks!",
      point_value: 10,
      receive_id: 59,
      received_name: "Bo Jackson",
      send_id: 43,
      send_name: "Deion Sanders"
    }
  ]}
]

export default mockGroupTransactions;