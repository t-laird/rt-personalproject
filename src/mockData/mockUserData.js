export const mockUserData = {
  authrocket_id: "usr_0vXWX9jSRkbUjLGsOb9S8r",
  created_date: "2017-11-18T18:34:30.017Z",
  email: "cpavin@usopen.com",
  group_id: 1,
  name: "Corey Pavin",
  user_id: 95
};

