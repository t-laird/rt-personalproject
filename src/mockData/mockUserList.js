export const mockUserList = [
  {
    authrocket_id: "jfnxcio42393",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "abcf@gmail.com",
    group_id: 1,
    name: "Mike Jenkins",
    user_id: 2
  },
  {
    authrocket_id: "lksdfjn23413",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "qwi123@gmail.com",
    group_id: 1,
    name: "Josh Brown",
    user_id: 3
  },
  {
    authrocket_id: "xmcv3941nv",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "293xcnv@gmail.com",
    group_id: 1,
    name: "Alan Michaelson",
    user_id: 4
  },
  {
    authrocket_id: "3490xxa1j",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "nc-1n23a@gmail.com",
    group_id: 1,
    name: "Sarah Hilman",
    user_id: 5    
  } 
];