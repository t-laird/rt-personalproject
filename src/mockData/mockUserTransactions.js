const mockUserTransactions = [
  {
    received: [
      {
        'send_id': 5,
        'receive_id': 8,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 1,
        'receive_id': 8,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ],
    sent: [
      {
        'send_id': 8,
        'receive_id': 4,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 8,
        'receive_id': 3,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ]
  }
];

export default mockUserTransactions;

