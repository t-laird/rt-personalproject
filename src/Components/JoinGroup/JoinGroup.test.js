import { JoinGroup, mapStateToProps, mapDispatchToProps } from './JoinGroup';
import React from 'react';
import { shallow } from 'enzyme';
import { mockUserData } from '../../mockData/mockUserData';
import { mockGroupData } from '../../mockData/mockGroupData';

global.localStorage = {
  getItem: () => null
};

describe('Join group tests', () => {
  window.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve([mockUserData])
    })
  );

  it('should match the snapshot when NO user is logged in', () => {
    const renderedApp = shallow(<JoinGroup group={{}} user={{}} />);

    expect(renderedApp).toMatchSnapshot();
  });

  it('should match the snapshot when a user is logged in', () => {
    const renderedApp = shallow(<JoinGroup group={ mockGroupData } user={ mockUserData } />);

    expect(renderedApp).toMatchSnapshot();
  });

  it('should update state on input', () => {
    const renderedApp = shallow(<JoinGroup group={{}} user={{}} />);
    const mockEvent = { target: {name: 'passphrase', value: 'abc'} };

    renderedApp.instance().handleChange(mockEvent);
    expect(renderedApp.state('passphrase')).toEqual('abc');
  });

  it('should display a message if joining the same group you are currently in', () => {
    const renderedApp = shallow(<JoinGroup user={{group_id: 5}} group={{group_id: 5, group_passphrase: '12345'}} />);
    const mockEvent = {preventDefault: jest.fn()};
    const expectedMessage = "You're already in that group!";
    renderedApp.setState({
      passphrase: '12345'
    });

    renderedApp.instance().joinGroup(mockEvent);

    expect(renderedApp.state('message').props.children).toEqual(expectedMessage);
  });
  it('should render no message if state says to hide message', () => {
    const renderedApp = shallow(<JoinGroup user={{group_id: 5}} group={{group_id: 5, group_passphrase: '12345'}} />);
    renderedApp.setState({
      hideintro: true
    });
    const messageResponse = renderedApp.instance().groupPageMessage();
  
    expect(messageResponse).toEqual(null);
  });

  it('should render a current group message if in a group currently', () => {
    const renderedApp = shallow(<JoinGroup user={{group_id: 5}} group={{group_id: 5, group_name:'TestGroup', group_passphrase: '12345'}} />);
    const messageResponse = renderedApp.instance().groupPageMessage();
    expect(messageResponse.props.children[0]).toContain('You are a member of:');
  });

  it('should render a message indicating that the user is already in a group if appropriate', () => {
    const renderedApp = shallow(<JoinGroup user={{group_id: 5}} group={{group_id: 5, group_passphrase: '12345'}} />);
    const messageResponse = renderedApp.instance().groupPageMessage();

    expect(messageResponse.props.children).toContain('You are already a member of a group!');    
  });

  it('should indicate that the user should log in if there is no user in store', () => {
    const renderedApp = shallow(<JoinGroup user={{}} group={{group_passphrase: '12345'}} />);
    const messageResponse = renderedApp.instance().groupPageMessage();

    expect(messageResponse.props.children[1]).toContain('to join a group or to see your group status!');
  });
});




describe('mapStateToProps tests', () => {
  it('should get user and group info from store', () => {
    const mockStore = {
      User: {
        authrocket_id: "usr_0vXXFrAmpjSWXnr7n2x2m8",
        created_date: "2017-11-18T18:34:30.017Z",
        email: "t6r6l5@gmail.com",
        group_id: '19vns',
        user_id: 20,
        name: "Thomas Laird"
      },
      Group: {
        administrator_id: 957,
        created_date: "2017-10-18T18:34:30.017Z",
        group_id: 15,
        group_name: null,
        group_passphrase: "19vns",
        weekly_points: 1000
      }
    };

    const result = mapStateToProps(mockStore);
    expect(result.user).toEqual(mockStore.User);
    expect(result.group).toEqual(mockStore.Group);
  });
});

describe('mapDispatchToProps tests', () => {
  it('should call dispatch when updateUser is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = {user_name:'Tom', user_id: 3};
    const result = mapDispatchToProps(mockDispatch);
    
    result.updateUser(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when updateUserTransactions is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = [{sent: [], received: []}];
    const result = mapDispatchToProps(mockDispatch);

    result.updateUserTransactions(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when updateGroupTransactions is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = [{transactions: []}];
    const result = mapDispatchToProps(mockDispatch);

    result.updateGroupTransactions(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when updateUserList is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = [{name: 'ann'}, {name: 'mike'}];
    const result = mapDispatchToProps(mockDispatch);

    result.updateUserList(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when updateGroup is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = {group_name: 'Test group'};
    const result = mapDispatchToProps(mockDispatch);

    result.updateGroup(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });
});


