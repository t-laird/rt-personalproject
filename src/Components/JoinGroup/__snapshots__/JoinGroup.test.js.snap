// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join group tests should match the snapshot when NO user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JoinGroup
    group={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="join-container"
>
          <h3>
                    <Link
                              replace={false}
                              to="/login"
                    >
                              Login
                    </Link>
                     to join a group or to see your group status!
          </h3>
          <form>
                    <input
                              name="passphrase"
                              onChange={[Function]}
                              placeholder="passphrase"
                              value=""
                    />
                    <button
                              onClick={[Function]}
                    >
                              SUBMIT
                    </button>
          </form>
</div>,
        <div
          className="link-holder"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="create-group-link"
                    to="/creategroup"
          >
                    CREATE NEW GROUP
          </NavLink>
</div>,
      ],
      "className": "join-group-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              <Link
                            replace={false}
                            to="/login"
              >
                            Login
              </Link>
               to join a group or to see your group status!
</h3>,
            <form>
              <input
                            name="passphrase"
                            onChange={[Function]}
                            placeholder="passphrase"
                            value=""
              />
              <button
                            onClick={[Function]}
              >
                            SUBMIT
              </button>
</form>,
            null,
          ],
          "className": "join-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/login"
>
                  Login
</Link>,
                " to join a group or to see your group status!",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              " to join a group or to see your group status!",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  name="passphrase"
                  onChange={[Function]}
                  placeholder="passphrase"
                  value=""
/>,
                <button
                  onClick={[Function]}
>
                  SUBMIT
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "passphrase",
                  "onChange": [Function],
                  "placeholder": "passphrase",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SUBMIT",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "SUBMIT",
                "type": "button",
              },
            ],
            "type": "form",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="create-group-link"
            to="/creategroup"
>
            CREATE NEW GROUP
</NavLink>,
          "className": "link-holder",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "CREATE NEW GROUP",
            "className": "create-group-link",
            "to": "/creategroup",
          },
          "ref": null,
          "rendered": "CREATE NEW GROUP",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="join-container"
>
            <h3>
                        <Link
                                    replace={false}
                                    to="/login"
                        >
                                    Login
                        </Link>
                         to join a group or to see your group status!
            </h3>
            <form>
                        <input
                                    name="passphrase"
                                    onChange={[Function]}
                                    placeholder="passphrase"
                                    value=""
                        />
                        <button
                                    onClick={[Function]}
                        >
                                    SUBMIT
                        </button>
            </form>
</div>,
          <div
            className="link-holder"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="create-group-link"
                        to="/creategroup"
            >
                        CREATE NEW GROUP
            </NavLink>
</div>,
        ],
        "className": "join-group-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <Link
                                replace={false}
                                to="/login"
                >
                                Login
                </Link>
                 to join a group or to see your group status!
</h3>,
              <form>
                <input
                                name="passphrase"
                                onChange={[Function]}
                                placeholder="passphrase"
                                value=""
                />
                <button
                                onClick={[Function]}
                >
                                SUBMIT
                </button>
</form>,
              null,
            ],
            "className": "join-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/login"
>
                    Login
</Link>,
                  " to join a group or to see your group status!",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                " to join a group or to see your group status!",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    name="passphrase"
                    onChange={[Function]}
                    placeholder="passphrase"
                    value=""
/>,
                  <button
                    onClick={[Function]}
>
                    SUBMIT
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "passphrase",
                    "onChange": [Function],
                    "placeholder": "passphrase",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SUBMIT",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "SUBMIT",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="create-group-link"
              to="/creategroup"
>
              CREATE NEW GROUP
</NavLink>,
            "className": "link-holder",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "CREATE NEW GROUP",
              "className": "create-group-link",
              "to": "/creategroup",
            },
            "ref": null,
            "rendered": "CREATE NEW GROUP",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Join group tests should match the snapshot when a user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JoinGroup
    group={
        Object {
            "administrator_id": 95,
            "created_date": "2017-10-18T18:34:30.017Z",
            "group_id": 1,
            "group_name": "Turing",
            "group_passphrase": "9139z",
            "weekly_points": 100,
          }
    }
    user={
        Object {
            "authrocket_id": "usr_0vXWX9jSRkbUjLGsOb9S8r",
            "created_date": "2017-11-18T18:34:30.017Z",
            "email": "cpavin@usopen.com",
            "group_id": 1,
            "name": "Corey Pavin",
            "user_id": 95,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="join-container"
>
          <h3>
                    You are a member of: 
                    <Link
                              replace={false}
                              to="/group"
                    >
                              Turing
                    </Link>
                    .  To switch groups, enter new passphrase below.
          </h3>
          <form>
                    <input
                              name="passphrase"
                              onChange={[Function]}
                              placeholder="passphrase"
                              value=""
                    />
                    <button
                              onClick={[Function]}
                    >
                              SUBMIT
                    </button>
          </form>
</div>,
        <div
          className="link-holder"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="create-group-link"
                    to="/creategroup"
          >
                    CREATE NEW GROUP
          </NavLink>
</div>,
      ],
      "className": "join-group-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              You are a member of: 
              <Link
                            replace={false}
                            to="/group"
              >
                            Turing
              </Link>
              .  To switch groups, enter new passphrase below.
</h3>,
            <form>
              <input
                            name="passphrase"
                            onChange={[Function]}
                            placeholder="passphrase"
                            value=""
              />
              <button
                            onClick={[Function]}
              >
                            SUBMIT
              </button>
</form>,
            null,
          ],
          "className": "join-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "You are a member of: ",
                <Link
                  replace={false}
                  to="/group"
>
                  Turing
</Link>,
                ".  To switch groups, enter new passphrase below.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "You are a member of: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Turing",
                  "replace": false,
                  "to": "/group",
                },
                "ref": null,
                "rendered": "Turing",
                "type": [Function],
              },
              ".  To switch groups, enter new passphrase below.",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  name="passphrase"
                  onChange={[Function]}
                  placeholder="passphrase"
                  value=""
/>,
                <button
                  onClick={[Function]}
>
                  SUBMIT
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "passphrase",
                  "onChange": [Function],
                  "placeholder": "passphrase",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SUBMIT",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "SUBMIT",
                "type": "button",
              },
            ],
            "type": "form",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="create-group-link"
            to="/creategroup"
>
            CREATE NEW GROUP
</NavLink>,
          "className": "link-holder",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "CREATE NEW GROUP",
            "className": "create-group-link",
            "to": "/creategroup",
          },
          "ref": null,
          "rendered": "CREATE NEW GROUP",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="join-container"
>
            <h3>
                        You are a member of: 
                        <Link
                                    replace={false}
                                    to="/group"
                        >
                                    Turing
                        </Link>
                        .  To switch groups, enter new passphrase below.
            </h3>
            <form>
                        <input
                                    name="passphrase"
                                    onChange={[Function]}
                                    placeholder="passphrase"
                                    value=""
                        />
                        <button
                                    onClick={[Function]}
                        >
                                    SUBMIT
                        </button>
            </form>
</div>,
          <div
            className="link-holder"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="create-group-link"
                        to="/creategroup"
            >
                        CREATE NEW GROUP
            </NavLink>
</div>,
        ],
        "className": "join-group-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                You are a member of: 
                <Link
                                replace={false}
                                to="/group"
                >
                                Turing
                </Link>
                .  To switch groups, enter new passphrase below.
</h3>,
              <form>
                <input
                                name="passphrase"
                                onChange={[Function]}
                                placeholder="passphrase"
                                value=""
                />
                <button
                                onClick={[Function]}
                >
                                SUBMIT
                </button>
</form>,
              null,
            ],
            "className": "join-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "You are a member of: ",
                  <Link
                    replace={false}
                    to="/group"
>
                    Turing
</Link>,
                  ".  To switch groups, enter new passphrase below.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "You are a member of: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Turing",
                    "replace": false,
                    "to": "/group",
                  },
                  "ref": null,
                  "rendered": "Turing",
                  "type": [Function],
                },
                ".  To switch groups, enter new passphrase below.",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    name="passphrase"
                    onChange={[Function]}
                    placeholder="passphrase"
                    value=""
/>,
                  <button
                    onClick={[Function]}
>
                    SUBMIT
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "passphrase",
                    "onChange": [Function],
                    "placeholder": "passphrase",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SUBMIT",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "SUBMIT",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="create-group-link"
              to="/creategroup"
>
              CREATE NEW GROUP
</NavLink>,
            "className": "link-holder",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "CREATE NEW GROUP",
              "className": "create-group-link",
              "to": "/creategroup",
            },
            "ref": null,
            "rendered": "CREATE NEW GROUP",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
